stories/utils/generate-widget-stories.js

```diff
@@ -158,6 +173,29 @@ export function generateReportBasedWidgetStories( {
 		},
 	};
 
+	// custom variants.
+	const customVariants = {};
+	Object.keys( additionalVariants ).forEach( ( name ) => {
+		const { data: variantData, options: variantOptions } = additionalVariants[ name ];
+		customVariants[ name ] = ( { dispatch } ) => {
+			if ( Array.isArray( variantOptions ) ) {
+				options.forEach( ( option, index ) => {
+					dispatch( datastore ).receiveGetReport( variantData[ index ], { options: option } );
+				} );
+			} else {
+				dispatch( datastore ).receiveGetReport( variantData, { options: variantOptions } );
+			}
+			const { [ name ]: callback } = additionalVariantCallbacks;
+			if ( callback ) {
+				callback( dispatch, data, options );
```

Comment:
Nit-picking, but I think this could be simplified to
```suggestion
			if ( additionalVariantCallbacks[ name ] ) {
				additionalVariantCallbacks[ name ]( dispatch, data, options );
```